#layout
uint32 sample_num
# data
float32[] alphas
